Does downloadApolloSchema support android product flavor?

**Question.**
In my project, I have set up both buildType and product flavors e.g.

```
buildTypes {
    getByName("release") {
      isMinifyEnabled = true
      proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
      isDebuggable = false
    }
    getByName("debug") {
      applicationIdSuffix(".debug")
      versionNameSuffix("-debug")
      isDebuggable = true
    }
  }

  flavorDimensions("mode")
  productFlavors {
    create("local") {
      dimension = "mode"
      applicationIdSuffix(".local")
      versionNameSuffix("-local")
      buildConfigField("String", "GRAPHQL_API", "\"https://local.abc.io/graphql\"")
    }
    create("staging") {
      dimension = "mode"
      applicationIdSuffix(".staging")
      versionNameSuffix("-staging")
      buildConfigField("String", "GRAPHQL_API", "\"https://staging.abc.io/graphql\"")
    }
  }
```
When I ran 
```
./gradlew downloadApolloSchema --endpoint="..." --schema="..."
```

I got
```
> Task :app:downloadApolloSchema FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:downloadApolloSchema'.
> Cannot find compilation unit, check variant and service.
  Possible services:
  service
  Possible variants:
  localDebug
  localDebugAndroidTest
  localDebugUnitTest
  localRelease
  localReleaseUnitTest
```
I can't find a way to resolve this error. If I remove the product flavor {}, I am able to download the `schema.json` again. I'm wondering if there are any workarounds for this issue.

--- a/apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/ApolloDownloadSchemaCliTask.kt
+++ b/apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/ApolloDownloadSchemaCliTask.kt
@@ -90,7 +90,7 @@ abstract class ApolloDownloadSchemaCliTask : DefaultTask() {
         return@filter true
       }
 
-      it.variantName == "main" || it.variantName == "release"
+      it.variantName == "main" || it.variantName.toLowerCase().contains("release")
     }
 
     val compilationUnit = if (service.isPresent) {
@@ -102,15 +102,9 @@ abstract class ApolloDownloadSchemaCliTask : DefaultTask() {
       candidates.firstOrNull()
     }
 
-    check(compilationUnit != null) {
-      val services = compilationUnits.map { it.serviceName }.distinct().sorted().joinToString("\n")
-      val variants = compilationUnits.map { it.variantName }.distinct().sorted().joinToString("\n")
-      "Cannot find compilation unit, check variant and service.\nPossible services:\n$services\nPossible variants:\n$variants"
-    }
+    val compilerParams = compilationUnit?.resolveParams(project)?.first
 
-    val (compilerParams, _) = compilationUnit.resolveParams(project)
-
-    var endpointUrl = endpoint.orProperty("endpoint") ?: compilationUnit.service.introspection?.endpointUrl?.get()
+    var endpointUrl = endpoint.orProperty("endpoint") ?: compilationUnit?.service?.introspection?.endpointUrl?.get()
 
     val schema = schema.orProperty("schema")?.let {
       if (schema.isPresent) {
@@ -118,7 +112,8 @@ abstract class ApolloDownloadSchemaCliTask : DefaultTask() {
       } else {
         project.file(it) // relative to project. This is not super consistent but 🤷‍
       }
-    } ?: compilerParams.schemaFile.asFile.get()
+    } ?: compilerParams?.schemaFile?.asFile?.get()
+      ?: throw IllegalArgumentException("ApolloGraphQL: cannot determine where to save the schema. Specify --schema or --service")
 
     val headersProp = project.findProperty("com.apollographql.apollo.headers") as? String
     val headers = when {

